### DISCUSSION TITLE
ao3downloader.py missing?

### DISCUSSION THREAD

**Discussion by fxgcxrx** [nianeyna/ao3downloader#53] [General]
I've been trying to get this to work for a while but as a new Windows user (got my laptop yesterday!) and as someone who knows nothing about Python, I can now pronounce myself Thoroughly Confused. I've downloaded and redownloaded Python and made sure the "add to path" thing is selected. But now when I go open the CMD file, this is what I get.
```

Error: [WinError 5] Access is denied: 'C:\\Windows\\System32\\venv'
The system cannot find the path specified.
Requirement already satisfied: pip in c:\users\vivian\appdata\local\programs\python\python39\lib\site-packages (22.2.2)
ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements.txt'
python: can't open file 'C:\Windows\System32\ao3downloader.py': [Errno 2] No such file or directory
Press any key to continue . . .
```
Apparently ao3downloader.py missing is the problem (or one of the problems)? It's right there in the folder I downloaded so I can't see why Windows Command can't find it. 

I'm stuck! Any advice?

**Comment by nianeyna** [comment]
Hm. Haven't seen this one before, but a quick google suggests it _might_ be because a bug that can happen when you try to create a python virtual environment in a folder that has spaces in the folder path. Try moving the script folder to the root of your `C:/` drive and see if that helps? If that doesn't help - or doesn't make sense to you - let me know.

**Reply by fxgcxrx** [reply]
Eek, I'm sorry, I didn't understand that at all. Which one is the script folder? What is the root of my c:/ drive?

**Reply by nianeyna** [reply]
No worries! The "script folder" is the folder containing ao3downloader.py - what you got after you unzipped the zip file that you downloaded from github. It's probably called "ao3downloader-main". `C:\` is the root of your folder system - pretty much everything saved on your computer has a file path that starts with `C:\`. You can find it (on windows 10 at least - if you're on windows 11 it might be slightly different) by opening windows file explorer, then clicking on "This PC" on the left. Then under "Devices and drives" you should see something with "C:" in the name. Double-click on that and it'll open a folder - that's the root folder of your C drive. What you want to do is move the "ao3downloader-main" folder into that folder (probably by dragging it from another windows file explorer window), then try to run the script again from there. I hope that made sense this time 😅 (Incidentally - the reason I'm suggesting C:\ specifically is because I know for sure that's a folder that exists on your computer and doesn't contain any spaces in the path, not because I necessarily recommend always saving the script there.)

**Reply by fxgcxrx** [reply]
Ok, did that, thank you! This is the new result from clicking the CMD file.
`'venv\Scripts\activate' is not recognized as an internal or external command,
operable program or batch file.
Traceback (most recent call last):
  File "C:\ao3downloader-main\ao3downloader.py", line 3, in <module>
    from ao3downloader.actions import ao3download
  File "C:\ao3downloader-main\ao3downloader\actions\ao3download.py", line 1, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'
Press any key to continue . . .
`

**Reply by nianeyna** [reply]
Is there a folder called "venv" in the same folder with the script? If so - try deleting that folder, then running the script again.

**Reply by fxgcxrx** [reply]
Argh, for some reason everytime I try to delete the "venv" folder, then try running the CMD file, then click back to the main file, "venv" is there again! I tried to follow [these](https://answers.microsoft.com/en-us/windows/forum/all/deleted-folders-keep-re-appearing-on-windows-10/4e63b399-920b-4912-927d-1a62d3c34591) instructions to no avail. Sorry to keep pestering you with questions! Have you ever had *this* problem before? Thank you for all your help!

**Reply by nianeyna** [reply]
oh my gosh. I'm so sorry. it's supposed to do that. one of the things the cmd file does is generate that folder - it's just that sometimes it gets stuck in a half-finished state and needs to be regenerated, that's why I had you delete it. not because it's not supposed to be there at all! you only needed to delete it one time - if it's still throwing errors after you did that, then a corrupted venv probably isn't the issue. What's the error message you're getting now? Is it still the "not recognized as a command" one?

**Reply by ryebone** [reply]
If you're still having trouble, I may have the solution. I was getting the same errors. Putting the entirety of the ao3downloader-main folder into the C:\Windows\System 32 folder (so it's no longer grouped in that main folder anymore), deleting the venv file, and then running the cmd as an administrator has fixed it for me. Running just fine for now. I think the script was having problems finding where the locations were.
